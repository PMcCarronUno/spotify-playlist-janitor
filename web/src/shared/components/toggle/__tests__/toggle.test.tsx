import { fireEvent, getByTestId, render } from "@testing-library/react";

import { Toggle } from "..";

describe("<Toggle />", () => {
  let container: HTMLElement;
  let props: any;
  let mockOnChange = jest.fn();

  beforeEach(() => {
    props = {
      label: "Toggle Example",
      onChange: mockOnChange,
      checked: false,
      id: "test-toggle",
      "data-testid": "test-toggle",
    };
    ({ container } = render(<Toggle {...props} />));
  });

  it("should render Toggle component", () => {
    expect(container.firstChild).toMatchSnapshot();
  });

  it("should render Toggle component toggled", () => {
    ({ container } = render(<Toggle {...{ ...props, checked: true }} />));
    expect(container.firstChild).toMatchSnapshot();
  });

  it("should fire onChange when clicked", () => {
    const input = getByTestId(container, "test-toggle");
    fireEvent.click(input);
    expect(mockOnChange).toHaveBeenCalled();
  });
});
