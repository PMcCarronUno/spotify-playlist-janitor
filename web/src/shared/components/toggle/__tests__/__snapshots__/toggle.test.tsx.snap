// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Toggle /> should render Toggle component 1`] = `
<label
  class="sc-beyTiQ lbQwve"
>
  <span>
    Toggle Example
  </span>
  <input
    class="sc-dmyDmy dfyiZa"
    data-testid="test-toggle"
    id="test-toggle"
    type="checkbox"
  />
  <div
    class="sc-guDMob kswJfi"
  />
</label>
`;

exports[`<Toggle /> should render Toggle component toggled 1`] = `
<label
  class="sc-beyTiQ lbQwve"
>
  <span>
    Toggle Example
  </span>
  <input
    checked=""
    class="sc-dmyDmy dfyiZa"
    data-testid="test-toggle"
    id="test-toggle"
    type="checkbox"
  />
  <div
    class="sc-guDMob kswJfi"
  />
</label>
`;
